chapters/functions.tex

```diff
@@ -2365,10 +2365,29 @@ \subsection{Annotations for External Functions}\label{annotations-for-external-l
   The {\lstinline!annotation(Library={"libraryName1", "libraryName2"})!}, used by the linker to include the library files where the compiled external function is available and additional libraries used to implement it.
   For shared libraries it is recommended to include all non-system libraries in this list.
 \item
-  The {\lstinline!annotation(Include="includeDirective")!}\annotationindex{Include}, used to include source files needed for calling the external function in the code generated by the Modelica compiler.
+  The {\lstinline!annotation(Include="insertedCode")!}\annotationindex{Include}, used to insert function prototypes or definitions needed for calling the external function in the code generated by the Modelica compiler.
+  When generating a call to the external function, the \lstinline!"insertedCode"! shall be present at the top level somewhere before the point of the call (similar to where include directives are typically placed).
+  The \lstinline!Include! annotation shall be used in such a way that each external function can be handled in a separate translation unit.
+  In particular, different external functions must not have \lstinline!Include! annotations providing exported definitions of the same funciton symbol to avoid linking errors.
+
+  A deprecated feature is that if multiple \lstinline!Include! annotations -- possibly coming from different external functions -- have identical content, the tool shall not incude this content more than once in any translation unit.
+  In case calls to several external functions are generated in the same translation unit, the \lstinline!Include! annotations of the different functions must not define the same function -- except when relying on the deprecated behavior.
+
   The included code should be valid C89 code.
+
   \begin{nonnormative}
-  Examples of files that can be included are header files or source files that contain the functions referenced in the external function declaration.
+  For an external function declaration calling the external function \lstinline[language=C]!myfoo!, examples of \lstinline!"insertedCode"! include:
+  \begin{itemize}
+  \item
+    An \lstinline[language=C]!#include! directive including a header file with a prototype for \lstinline[language=C]!myfoo!.
+  \item
+    An \lstinline[language=C]!#include! directive including a source file with a \lstinline[language=C]!static! definition of \lstinline[language=C]!myfoo!.
+    Include guards should be used (either in the \lstinline!Include! annotation or in the source file) to avoid relying on the deprecated feature that shall not include multiple copies in the same translation unit.
```

Comment:
Reformulate "to avoid relying on the deprecated feature that shall not include multiple copies in the same translation unit." since it doesn't parse

Reply:
Inserted _tools_ so that the sentence now parses.  Resolving.